File: engine/tests/test_ws.py
import asyncio
import httpx
import websockets
import json

BASE_URL = 'http://127.0.0.1:8000'
BASE_WS = 'ws://127.0.0.1:8000'
API_KEY = 'SLINGSHOT_INTERNAL_V6'

async def test_ws():
    print("1. Fetching token...")
    try:
        async with httpx.AsyncClient() as client:
            r = await client.get(f"{BASE_URL}/api/v1/auth/token?api_key={API_KEY}")
            if r.status_code != 200:
                print("Failed to get token:", r.status_code)
                return
            token = r.json()["token"]
            print("Token retrieved successfully:", token)

        ws_url = f"{BASE_WS}/api/v1/stream/BTCUSDT?token={token}&interval=15m"
        print(f"\n2. Connecting to WebSocket: {ws_url}...")
        async with websockets.connect(ws_url) as ws:
            print("Connected! Waiting for 3 messages to verify stream integrity...")
            for i in range(3):
                msg = await ws.recv()
                data = json.loads(msg)
                print(f"\n[MSG #{i+1}] Type: {data.get('type')}")
                if data.get('type') == 'smc_data':
                    print("Dynamic SMC Order Blocks:")
                    print(json.dumps(data.get('data', {}).get('order_blocks'), indent=2))
                elif data.get('type') == 'radar_update':
                    print(f"Radar assets count: {len(data.get('data', []))}")
                elif data.get('type') == 'history':
                    print(f"Candles history count: {len(data.get('data', []))}")
                    
    except Exception as e:
        import traceback
        print("Error during test:")
        traceback.print_exc()

if __name__ == '__main__':
    asyncio.run(test_ws())

File: engine/tests/test_ws.js
const WebSocket = require('ws');

const BASE_URL = 'http://127.0.0.1:8000';
const BASE_WS = 'ws://127.0.0.1:8000';
const API_KEY = 'slingshot_secure_key_v10'; // Default key

async function main() {
    console.log("1. Fetching auth token...");
    try {
        const res = await fetch(`${BASE_URL}/api/v1/auth/token?api_key=${API_KEY}`);
        if (!res.ok) {
            console.error("Failed to fetch token. Status:", res.status);
            return;
        }
        const data = await res.json();
        const token = data.token;
        console.log("Token retrieved successfully:", token);

        console.log("\n2. Connecting to WebSocket stream for BTCUSDT...");
        const wsUrl = `${BASE_WS}/api/v1/stream/BTCUSDT?token=${token}&interval=15m`;
        const ws = new WebSocket(wsUrl);

        ws.on('open', () => {
            console.log("WebSocket connection established! Waiting for messages...");
        });

        let msgCount = 0;
        ws.on('message', (message) => {
            msgCount++;
            const parsed = JSON.parse(message);
            console.log(`\n[MSG #${msgCount}] Type: ${parsed.type}`);
            
            if (parsed.type === 'smc_data') {
                console.log("SMC Data Payload keys:", Object.keys(parsed.data || {}));
                console.log("Order Blocks:", JSON.stringify(parsed.data?.order_blocks, null, 2));
                console.log("FVGs:", JSON.stringify(parsed.data?.fvgs, null, 2));
            } else if (parsed.type === 'radar_update') {
                console.log("Radar update size:", parsed.data?.length);
            } else if (parsed.type === 'tactical_update') {
                console.log("Tactical update asset:", parsed.data?.asset);
                console.log("Tactical update signals:", parsed.data?.signals);
            }
            
            if (msgCount >= 5) {
                console.log("\nTest completed. Closing connection...");
                ws.close();
                process.exit(0);
            }
        });

        ws.on('error', (err) => {
            console.error("WS Error:", err);
        });

        ws.on('close', (code, reason) => {
            console.log(`WebSocket closed. Code: ${code}, Reason: ${reason.toString()}`);
        });

    } catch (e) {
        console.error("Error during execution:", e);
    }
}

main();
